pkg/scheduler/preemption/preemption.go

```diff
@@ -376,8 +377,17 @@ func workloadFits(wlReq cache.FlavorResourceQuantities, cq *cache.ClusterQueue,
 						}
 					}
 				}
-				if cq.Cohort != nil && cohortResUsage[rName]+rReq > cohortResRequestable[rName] {
-					return false
+
+				if cq.Cohort != nil {
+					requestableQuota := cohortResRequestable[rName]
+					if features.Enabled(features.LendingLimit) && flvQuotas.Resources[rName].LendingLimit != nil {
+						// If LendingLimit is not nil here, we need to recalculate cohort.RequestableResources,
+						// because we are borrowing from the cohort now, need to count our own nominal quota in.
+						requestableQuota += cq.GetGuaranteedQuota(flvQuotas.Name, rName)
+					}
```

Comment:
This whole logic appears at least 3 times. Please put all of it in a function.

Reply:
Done.